packages/repack/src/modules/ScriptManager/types.ts

```diff
@@ -88,6 +88,25 @@ export interface ScriptLocator {
    * Setting this flat to `false`, disables that behavior.
    */
   cache?: boolean;
+  /**
+   * Function called before loading or getting from the cache and after resolving the script locator.
+   *
+   * When `true` is returned, the script will be loaded from the network.
+   * When `false` is returned, the script will be loaded from the cache.
+   *
+   * @param scriptId Id of the script to resolve.
+   * @param caller Name of the calling script - it can be for example: name of the bundle, chunk or container.
+   * @param isOutdated Boolean indicating whether the script cache is outdated or not. It's `true` when the script
+   * cache is outdated and `false` when the script cache is up to date or there is no cache for the script.
+   * Outdated cache means that the script was previously downloaded and put into cache,
+   * but the script locator data (method, url, query, headers, or body) has changed since then.
+   * @returns Boolean indicating whether the script should be loaded or not
+   */
+  shouldUpdateScript?: (
+    scriptId?: string,
+    caller?: string,
+    isOutdated?: boolean
```

Comment:
I have a hard time wrapping my head around the name `isOutdated`, especially in the context of how it is used in the description of the PR, specifically this part:

```
if (!isOutdated) {
        return true;
      }
```

It just looks weird to return `true` from function named `shouldUpdateScript` if _something_ is NOT outdated (`!isOutdated`). It just reads weird. I think if you would change the parameter name to `isCacheOutdated` or `isScriptCacheOutdated` it will be much clearer. It makes even more sense, because the actual value of that parameter is the result of evaluating `return this.checkIfCacheDataOutdated(cachedData)` and it reads like `return is the cache outdated`